### PR TITLE
Add page for Remote buckets

### PR DESCRIPTION
**Pull request by adfost**
Adding a new page to the menu to view remote buckets.
Screen Shot 2021-07-01 at 3 25 23 PM

### CODE REVIEW

**Comment by Alevsk** [comment]
@adfost does this new `/remote-buckets` page needs a new entry on the `acl.go` file ?

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+import AddBucket from "./AddBucket";
+import DeleteBucket from "./DeleteBucket";
+import PageHeader from "../../Common/PageHeader/PageHeader";
+import BulkReplicationModal from "./BulkReplicationModal";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    seeMore: {
+      marginTop: theme.spacing(3),
+    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:52]: This style is not needed

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+    paper: {
+      display: "flex",
+      overflow: "auto",
+      flexDirection: "column",
+    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:57]: This style is not needed

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+
+    addSideBar: {
+      width: "320px",
+      padding: "20px",
+    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:62]: This style is not needed

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+      paddingRight: theme.spacing(0),
+    },
+    minTableHeader: {
+      color: "#393939",
+      "& tr": {
+        "& th": {
+          fontWeight: "bold",
+        },
+      },
+    },
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:74]: These styles are not needed

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    ...actionsTray,
+    ...searchField,
+    ...containerForHeader(theme.spacing(4)),
+  });
+
+interface IListBucketsProps {
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:80]: Would be better to rename this as `IListRemoteBucketsProps`

```suggestion
interface IListRemoteBucketsProps {
```

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+  };
+
+  return (
+    <Fragment>
+      {addBucketModalOpen && (
+        <AddBucket
+          open={addBucketModalOpen}
+          closeModalAndRefresh={closeAddModalAndRefresh}
+        />
+      )}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:238]: This is tied to local bucket creation. I don't think we need this modal here as we are talking of different scopes.

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 7]
```diff
@@ -0,0 +1,341 @@
...
+      {deleteOpen && (
+        <DeleteBucket
+          deleteOpen={deleteOpen}
+          selectedBucket={selectedBucket}
+          closeDeleteModalAndRefresh={(refresh: boolean) => {
+            closeDeleteModalAndRefresh(refresh);
+          }}
+        />
+      )}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:247]: In this case I don't think we can delete remote buckets from local environment

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 8]
```diff
@@ -0,0 +1,341 @@
...
+      {replicationModalOpen && (
+        <BulkReplicationModal
+          open={replicationModalOpen}
+          buckets={selectedBuckets}
+          closeModalAndRefresh={closeBulkReplicationModal}
+        />
+      )}
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:254]: This modal was meant for local buckets to be replicated to remote targets. This screen should list remote targets only so this is not needed here

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 9]
```diff
@@ -0,0 +1,341 @@
...
+      <PageHeader label={"Remote Buckets"} />
+      <Grid container>
+        <Grid item xs={12} className={classes.container}>
+          <Grid item xs={12} className={classes.actionsTray}>
+            <TextField
+              placeholder="Search Buckets"
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:260]:
```suggestion
              placeholder="Search Remote Buckets"
```

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 10]
```diff
@@ -0,0 +1,341 @@
...
+              <Button
+                variant="contained"
+                color="primary"
+                startIcon={<CreateIcon />}
+                onClick={() => {
+                  addBucketOpen(true);
+                }}
+              >
+                Create Bucket
+              </Button>
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:297]: These buttons are meant for local targets only, anything you do with these modals will not appear in this module

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 11]
```diff
@@ -0,0 +1,341 @@
...
+                  elementKey: "targetBucket",
+                },
+                {
+                  label: "Target URL",
+                  elementKey: "targetURL",
+                },
+              ]}
+              isLoading={loading}
+              records={filteredRecords}
+              entityName="RemoteBuckets"
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:319]:
```suggestion
              entityName="Remote Buckets"
```

File: portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx [hunk 12]
```diff
@@ -0,0 +1,341 @@
...
+  };
+
+  const tableActions = [
+    { type: "view", to: `/buckets`, sendOnlyId: true },
+    { type: "delete", onClick: confirmDeleteBucket, sendOnlyId: true },
+  ];
+
+  const displayParsedDate = (date: string) => {
+    return <Moment>{date}</Moment>;
+  };
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/ListBuckets/ListRemoteBuckets.tsx:190]: This is not used. Please remove

File: portal-ui/src/screens/Console/Buckets/RemoteBuckets.tsx [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+});
+
+const connector = connect(mapState, { setMenuOpen });
+
+const RemoteBuckets = () => {
+  return (
+    <Router history={history}>
+      <Switch>
+        <Route path="/remote-buckets/:bucketName/*" component={BucketDetails} />
+        <Route path="/remote-buckets/:bucketName" component={BucketDetails} />
```
Comment (bexsoft) [inline portal-ui/src/screens/Console/Buckets/RemoteBuckets.tsx:38]: You need to add these paths to endpoints.go.

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Comment by bexsoft** [comment]
Please remabe with master

**Pull request closed without merging**